### DISCUSSION TITLE
Refreshing display

### DISCUSSION THREAD

**Discussion by maknotmark** [symfony/symfony#50535] [Q&A]
I understand this is a relatively dumb question but please bear with me. 

When running this form alone on its route, the var dump is displayed after submit: 

Controller:
``` 
<?php
namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use App\Form\FormType;
use Doctrine\ORM\EntityManagerInterface;
use Symfony\Component\HttpFoundation\Request;

class FormController extends AbstractController
{
    #[Route('/app/form', name: 'app_form')]
    public function new(EntityManagerInterface $em, Request $request): Response
    {
        $form = $this->createForm(FormType::class);
        
        $form->handleRequest($request);
    
        if ($form->isSubmitted() && $form->isValid()) {
             var_dump($form->getData());
        }

        return $this->render('form/index_empty.html.twig', [
            'articleForm' => $form->createView(),
        ]);
    }       
}
```
AppForm Template:
```
  {{ form_start(articleForm) }}
      {{ form(articleForm) }}
      {{ dump(data) }}
  {{ form_end(articleForm) }}
```

Base.html.twig: 
```
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8">
        <title>{% block title %}Welcome!{% endblock %}</title>
        <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 128 128%22><text y=%221.2em%22 font-size=%2296%22>⚫️</text></svg>">
        {# Run `composer require symfony/webpack-encore-bundle` to start using Symfony UX #}
        {% block stylesheets %}
            {{ encore_entry_link_tags('app') }}
        {% endblock %}

        {% block javascripts %}
            {{ encore_entry_script_tags('app') }}
        {% endblock %}
    </head>
    <body>
        {% block body %}{% endblock %}
    {{ render('/app/form') }}
    </body>
</html>
```

Submitting on '/app/form' displays the vardump: 
![image](https://github.com/symfony/symfony/assets/122203779/209c130b-a75b-4050-9608-cef3ceee3e69)

Submitting on '/' does not display the vardump: 
![image](https://github.com/symfony/symfony/assets/122203779/2c0e8842-0301-4abe-a2b3-8ed90cd62541)

even though the POST is getting sent as: 
![image](https://github.com/symfony/symfony/assets/122203779/350baf84-260f-4d37-a600-dc7857d1db38)

Can someone point me in the correct direction, my understanding is that i would have to somehow do an ajax call on the POST data?

**Comment by ghost** [comment]
I don't understand your problem. Please clean up and maybe reword the description. Also lines with sentences disappear among all those pictures. Remember - the more familiar you description is the more chances someone will help you.

P.S. Nobody uses var_dump nowadays, use dump() from symfony/var-dumper.

**Reply by ghost** [reply]
Ok, after reading your description 10 times I think I know what's your problem. You `/` route renders `base.html.twig` and this template has a line `{{ render('/app/form) }}` and `/app/form` route has a `var_dump` line. `/` does not render var_dump output because twig's `render` function sends a GET request.

> Submitting on '/' does not display the vardump:

When submiting the form here you don't submit to `/` but to `/app/form` (this is a route where form data will be submitted).

**Reply by maknotmark** [reply]
Thanks for the speedy reply! 

regarding var_dump I was typing faster than i was thinking it's rendered with {{ dump(data) }} in the Twig Template. I think this is what you meant with your P.S. or is ther another dump() function im not aware of?

> because twig's render function sends a GET request
this explains a lot!! Thanks for the clarification, i knew it was something fundamental that I was missing. 

I've tried reading up on the documentation but i can't seem to work around it. Would setting it to a POST request solve the issue?
